propolis/src/block.rs

```diff
@@ -25,13 +28,19 @@ pub enum BlockResult {
     Unsupported,
 }
 
+/// Trait indicating that a type may be used as a request to a block device.
 pub trait BlockReq: Send + Sync + 'static {
+    /// Type of operation being issued.
     fn oper(&self) -> BlockOp;
+    /// Offset within the block device, in bytes.
     fn offset(&self) -> usize;
+    /// Returns the next region of memory within a request to a block device.
     fn next_buf(&mut self) -> Option<GuestRegion>;
+    /// Signals to the guest that a block operation has been completed.
```

Comment:
It's really signalling to the driver emulation that the operation is complete.  It's up to the emulation how/if the guest should be notified.